camel-openmrs-fhir/src/test/java/org/openmrs/eip/fhir/spring/OpenmrsFhirConfigurationTest.java

```diff
@@ -14,22 +17,47 @@
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.TestPropertySource;
 
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import ca.uhn.fhir.rest.client.interceptor.BasicAuthInterceptor;
+
 /**
  * We have some slightly gnarly code to convert the {@code eip.fhir.resources} property to some
  * properties for the watcher. Here we just ensure that this gets configured correctly if the
  * {@link OpenmrsFhirConfiguration} is loaded.
  */
 @SpringBootTest
 @ContextConfiguration(classes = OpenmrsFhirConfiguration.class)
-@TestPropertySource(properties = { "eip.fhir.resources=Encounter", "openmrs.eip.log.level=DEBUG" })
+@TestPropertySource(properties = { "eip.fhir.resources=Encounter", "openmrs.eip.log.level=DEBUG",
+        "eip.fhir.serverUrl=http://localhost:8080/openmrs/ws/fhir2/R4", "eip.fhir.username=admin",
+        "eip.fhir.password=password" })
 public class OpenmrsFhirConfigurationTest {
 	
 	@Autowired
 	Environment environment;
 	
+	@Autowired
+	private IGenericClient openmrsFhirClient;
+	
 	@Test
 	void shouldConfigureContextProperly() {
 		assertThat(environment.getProperty(PROP_WATCHED_TABLES), equalTo(String.join(",", FhirResource.ENCOUNTER.tables())));
 		assertThat(environment.getProperty(PROP_EVENT_DESTINATIONS), equalTo(URI_FHIR_ROUTER));
 	}
+	
+	@Test
+	void shouldRegisterIGenericClientBean() {
+		assertNotNull(openmrsFhirClient);
+	}
+	
+	@Test
+	void shouldRegisterBasicAuthInterceptorWithCorrectUsernameAndPassword() {
+		// verify username and password are set correctly
+		assertEquals("admin", environment.getProperty("eip.fhir.username"));
+		assertEquals("password", environment.getProperty("eip.fhir.password"));
+		
+		var interceptors = openmrsFhirClient.getInterceptorService().getAllRegisteredInterceptors();
+		// verify BasicAuthInterceptor is registered
+		assertNotNull(interceptors);
+		assertTrue(interceptors.stream().anyMatch(BasicAuthInterceptor.class::isInstance));
+	}
```

Comment:
From  a pure unit test perspective, I don't think I like the way these tests are setup, because instead of calling the logic you're testing from the test method i.e. constructor, you're testing a pre-constructed bean by spring. I guess it's okay but I'd have preferred you call the constructor from the test method itself.